contracts/permissionless/OrderbookReserve.sol

```diff
@@ -0,0 +1,870 @@
+        require(maker != address(0));
+        require(amount < MAX_QTY);
+
+        require(kncToken.transferFrom(msg.sender, this, amount));
+
+        makerKnc[maker].freeKnc += uint128(amount);
+
+        KncFeeDeposited(maker, amount);
+
+        if (orderAllocationRequired(makerOrdersTokenToEth[maker])) {
+            require(allocateOrderIds(
+                makerOrdersTokenToEth[maker], /* freeOrders */
+                tokenToEthList.allocateIds(uint32(NUM_ORDERS)) /* firstAllocatedId */
+            ));
+        }
+
+        if (orderAllocationRequired(makerOrdersEthToToken[maker])) {
+            require(allocateOrderIds(
+                makerOrdersEthToToken[maker], /* freeOrders */
+                ethToTokenList.allocateIds(uint32(NUM_ORDERS))
+            ));
+        }
+    }
+
+    function withdrawToken(uint amount) public {
+
+        address maker = msg.sender;
+        uint makerFreeAmount = makerFunds[maker][token];
+
+        require(makerFreeAmount >= amount);
+
+        makerFunds[maker][token] -= amount;
+
+        require(token.transfer(maker, amount));
+    }
+
+    function withdrawEther(uint amount) public {
+
+        address maker = msg.sender;
+        uint makerFreeAmount = makerFunds[maker][ETH_TOKEN_ADDRESS];
+
+        require(makerFreeAmount >= amount);
+
+        makerFunds[maker][ETH_TOKEN_ADDRESS] -= amount;
+
+        maker.transfer(amount);
+    }
+
+    function withdrawKncFee(uint amount) public {
+
+        address maker = msg.sender;
+        uint makerFreeAmount = uint(makerKnc[maker].freeKnc);
+
+        require(makerFreeAmount >= amount);
+
+        require(uint(uint128(amount)) == amount);
+
+        makerKnc[maker].freeKnc -= uint128(amount);
+
+        require(kncToken.transfer(maker, amount));
+    }
+
+    function cancelTokenToEthOrder(uint32 orderId) public returns(bool) {
+        require(cancelOrder(false, orderId));
+        return true;
+    }
+
+    function cancelEthToTokenOrder(uint32 orderId) public returns(bool) {
+        require(cancelOrder(true, orderId));
+        return true;
+    }
+
+    event KncStakePerEthSet(uint currectStakeBps, uint newStakeBps, address feeBurnerContract, address sender);
+
+    function setStakePerEth(uint newStakeBps) public {
+
+        uint factor = 2;
+        uint burnPerWeiBps = (makerBurnFeeBps * feeBurnerContract.kncPerEthRatePrecision()) / PRECISION;
+
+        // old factor should be too small
+        require(kncStakePerEtherBps < factor * burnPerWeiBps);
+
+        // new value should be high enough
+        require(newStakeBps > factor * burnPerWeiBps);
+
+        // but not too high...
+        require(newStakeBps > (factor + 1) * burnPerWeiBps);
+
+        KncStakePerEthSet(kncStakePerEtherBps, newStakeBps, feeBurnerContract, msg.sender);
+        // Ta daaa
+        kncStakePerEtherBps = newStakeBps;
+    }
+
+    function getAddOrderHintTokenToEth(uint128 srcAmount, uint128 dstAmount) public view returns (uint32) {
```

Comment:
according to the function namings before it should be getAddTokenToEthOrderHint().
Also relevant for the functions below.